File: matlab-pipeline/NBSR_TIP12/NBSR_core/Eval/my_psnr.m
function [pv, mse] = my_psnr(img1,img2)
% PSNR Evaluation
%Image value should be in range [0 255]
%2010-1-18 15:28:32
%Haichao Zhang 
err =double( img1)-double(img2);
mse = sum(sum(err.^2))/ prod(size(img1));

%if(max(img1(:))>100)
    pv = 20*log10(255/sqrt(mse));
% else
%     pv = 20*log10(1/sqrt(mse));
% end

File: python-pipeline/eval/my_psnr.py
import numpy as np
from functools import reduce
from operator import mul
import math

def my_psnr(img1, img2):
    # Image value should be in range [0 255]
    err = np.substract(img1.astype(np.double), img2.astype(np.double))
    mse = sum((np.power(err, 2)).sum(axis=0) / reduce(mul, img1.shape))

    pv = 20*math.log10(255/math.sqrt(mse))

    return pv, mse